cmake/common_compiler_flags.cmake

```diff
@@ -8,11 +8,26 @@ flags look to each of the ``cmake/<platform>.cmake`` files.
 
 ]=======================================================================]
 #Generator Expression Helpers
-set( IS_CLANG "$<CXX_COMPILER_ID:Clang>" )
-set( IS_APPLECLANG "$<CXX_COMPILER_ID:AppleClang>" )
-set( IS_GNU "$<CXX_COMPILER_ID:GNU>" )
-set( IS_MSVC "$<CXX_COMPILER_ID:MSVC>" )
-set( NOT_MSVC "$<NOT:$<CXX_COMPILER_ID:MSVC>>" )
+function(helper_variables)
+    set( IS_CLANG      "$<CXX_COMPILER_ID:Clang>"       PARENT_SCOPE )
+    set( IS_APPLECLANG "$<CXX_COMPILER_ID:AppleClang>"  PARENT_SCOPE )
+    set( IS_GNU        "$<CXX_COMPILER_ID:GNU>"         PARENT_SCOPE )
+    set( IS_MSVC       "$<CXX_COMPILER_ID:MSVC>"        PARENT_SCOPE )
+    set( NOT_MSVC      "$<NOT:$<CXX_COMPILER_ID:MSVC>>" PARENT_SCOPE )
+    if( ${CMAKE_CXX_COMPILER_ID} STREQUAL Clang )
+        if( ${CMAKE_CXX_COMPILER_FRONTEND_VARIANT} STREQUAL MSVC )
+            set( IS_CLANG "0" PARENT_SCOPE )
+            set( IS_MSVC "1" PARENT_SCOPE )
+            set( NOT_MSVC "0" PARENT_SCOPE )
+        endif ()
+        # We dont yet support splitting the compiler/link command line interface.
+        # they have to be either both GNU or both MSVC
+#        if( ${CMAKE_CXX_COMPILER_LINKER_FRONTEND_VARIANT} STREQUAL MSVC )
+#            set( IS_MSVC "1" PARENT_SCOPE )
+#            set( IS_CLANG "0" PARENT_SCOPE )
+#        endif ()
```

Comment:
I think we could probably just delete this comment and commented code? I don't know if we can ever support using a mixed compiler and linker, and I'm not sure there's a good reason to anyway.

Reply:
Done.